qlib/TableMapper.qm

```diff
@@ -880,8 +993,15 @@ on_error table_mapper.rollback();
 
             # get return value
             hash rv = hbuf + rh;
-            # clear buffer but keep hash keys
-            map hbuf.$1 = (), hbuf.keyIterator();
+            if (exists rec_list_size) {
```

Comment:
`hbuf` is meant to always be in hash of lists format, even when used with input data in another format.  This allows us to have fast inserts with all sorts of input data.   It is an error to treat `hbuf` as a flat hash for example - it should always be a hash of lists with possible single values for constant data